### 2. BASE TAXA DE RISCO

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

A fonte dos dados foi recebida em formato planilha Excel. No próprio Excel, foi ajustado para criação de base em xls já ajustada para entidades em Goiás e municípios em Goiás. Foram ajustadas nomes colunas e retirados colunas mescladas.

In [2]:
df = pd.read_excel('Base_TR\BRASIL_TR_2020_21_ajustada_escola.xlsx')  

In [3]:
df.head(3)

CODIGO_UF  UF  CO_MUNICIPIO         NOME_MUNICIPIO  CO_ENTIDADE  \
0         52  GO       5200258  Águas Lindas de Goiás     52086038   
1         52  GO       5200258  Águas Lindas de Goiás     52104494   
2         52  GO       5200258  Águas Lindas de Goiás     52105750   

                              NOME_ENTIDADE  TAXA_RISCO_ENT  
0  COLEGIO ESTADUAL RAFAEL DE SOUZA BARBOSA               5  
1                              ESCOLA CEMYR               9  
2                           COLEGIO GRADUAL               6

In [4]:
# Total de escolas goianas na base de taxa de risco do INEP
df.shape

(126, 7)

126 entidades/escolas

#### Quantidade de escolas por Taxa de Risco?


In [5]:
d5 = []
d6 = []
d7 = []
d8 = []
d9 = []
d10 = []
d11 = []
d12 = []

for x in df['TAXA_RISCO_ENT']:
    if x == 5:
        d5.append(x)
    if x == 6:
        d6.append(x)
    if x == 7:
        d7.append(x)
    if x == 8:
        d8.append(x)
    if x == 9:
        d9.append(x)
    if x == 10:
        d10.append(x)
    if x == 11:
        d11.append(x)
    if x == 12:
        d12.append(x)
            
print("TR5 - quantidade:", len(d5))
print("TR6 - quantidade:", len(d6))
print("TR7 - quantidade:", len(d7))
print("TR8 - quantidade:", len(d8))
print("TR9 - quantidade:", len(d9))
print("TR10 - quantidade:", len(d10))
print("TR11 - quantidade:", len(d11))
print("TR12 - quantidade:", len(d12))

TR5 - quantidade: 67
TR6 - quantidade: 28
TR7 - quantidade: 13
TR8 - quantidade: 4
TR9 - quantidade: 7
TR10 - quantidade: 6
TR11 - quantidade: 1
TR12 - quantidade: 0


In [6]:
print("TR 5 ou mais:", 67 + 28 + 13 + 4 + 7 + 6 + 1)
print("TR 6 ou mais:", 28 + 13 + 4 + 7 + 6 + 1)
print("TR 7 ou mais:", 13 + 4 + 7 + 6 + 1)
print("TR 8 ou mais:", 4 + 7 + 6 + 1)
print("TR 9 ou mais:", 7 + 6 + 1)
print("TR 10 ou mais:", 6 + 1)
print("TR 11 ou mais:", 1)

TR 5 ou mais: 126
TR 6 ou mais: 59
TR 7 ou mais: 31
TR 8 ou mais: 18
TR 9 ou mais: 14
TR 10 ou mais: 7
TR 11 ou mais: 1


In [7]:
tab = pd.crosstab(index=df['TAXA_RISCO_ENT'], columns='count')

tab

col_0           count
TAXA_RISCO_ENT       
5                  67
6                  28
7                  13
8                   4
9                   7
10                  6
11                  1

#### Os valores acima serão referências para parâmetros de algoritmos de detecção de outlier.

#### Exemplo que como fazer extração de escolas 

In [8]:
df_tr_m9 = df.loc[df['TAXA_RISCO_ENT'] >= 9]

In [9]:
df_tr_m9.shape

(14, 7)

In [10]:
df_tr_m9

CODIGO_UF  UF  CO_MUNICIPIO          NOME_MUNICIPIO  CO_ENTIDADE  \
1           52  GO       5200258   Águas Lindas de Goiás     52104494   
23          52  GO       5204854    Campo Limpo de Goiás     52103293   
28          52  GO       5205109                 Catalão     52105768   
66          52  GO       5208707                 Goiânia     52102394   
70          52  GO       5208707                 Goiânia     52105571   
71          52  GO       5208707                 Goiânia     52105725   
72          52  GO       5208707                 Goiânia     52105792   
73          52  GO       5208707                 Goiânia     52105865   
76          52  GO       5208905                   Goiás     52003388   
97          52  GO       5215231               Novo Gama     52047172   
99          52  GO       5215231               Novo Gama     52105083   
113         52  GO       5220207  São Miguel do Araguaia     52070107   
122         52  GO       5221858     Valparaíso de Goiás     52103943   
124         52  GO       5221858     Valparaíso de Goiás     52105644   

                                         NOME_ENTIDADE  TAXA_RISCO_ENT  
1                                         ESCOLA CEMYR               9  
23                           ESCOLA GOTAS DE ESPERANCA              10  
28                                ESCOLA CASINHA FELIZ               9  
66   INSTITUTO DE EDUCACAO EM ARTES PROFESSOR GUSTA...              10  
70                                      ESCOLA DIG JOY               9  
71                            COLEGIO CONECTAR LTDA-ME               9  
72                   BERCARIO E ESCOLA SAGRADA FAMILIA              10  
73                                        ESCOLA ELLOS              10  
76                    ESCOLA ESTADUAL POVOADO SAO JOAO              10  
97                                      CAIC NOVO GAMA               9  
99                       ESCOLA E CRECHE AMOR PERFEITO              10  
113            ESCOLA ESPECIAL ZELI DE OLIVEIRA VILELA              11  
122                     ESCOLA EVANGELICA LAPIS NA MAO               9  
124                    ESCOLA ADVENTISTA DE VALPARAISO               9

In [11]:
df.dtypes

CODIGO_UF          int64
UF                object
CO_MUNICIPIO       int64
NOME_MUNICIPIO    object
CO_ENTIDADE        int64
NOME_ENTIDADE     object
TAXA_RISCO_ENT     int64
dtype: object

#### Base municípios goianos com taxa de risco para o INEP

In [12]:
df2 = pd.read_excel('Base_TR\BRASIL_TR_2020_21_ajustada_mun.xlsx')  

In [13]:
df2

CO_UF  UF  CO_MUNICIPIO            NOME_MUNICIPIO  TAXA_RISCO_MUN
0      52  GO       5200605     Alto Paraíso de Goiás               6
1      52  GO       5204003                Cabeceiras               6
2      52  GO       5204854      Campo Limpo de Goiás               7
3      52  GO       5205406                     Ceres               5
4      52  GO       5205521            Colinas do Sul               5
5      52  GO       5206909               Davinópolis               5
6      52  GO       5207600              Fazenda Nova               6
7      52  GO       5208301      Divinópolis de Goiás               5
8      52  GO       5209150               Gouvelândia               6
9      52  GO       5209804                 Hidrolina               5
10     52  GO       5212907                  Marzagão               6
11     52  GO       5213053           Mimoso de Goiás               5
12     52  GO       5213400                   Moiporá               6
13     52  GO       5214408                   Nazário               7
14     52  GO       5216403                   Paraúna               5
15     52  GO       5218052                 Porteirão               5
16     52  GO       5219407    Santa Rita do Araguaia               5
17     52  GO       5219738    Santo Antônio de Goiás               5
18     52  GO       5220108  São Luís de Montes Belos               5
19     52  GO       5221007         Taquaral de Goiás               7
20     52  GO       5221809                    Urutaí               5

In [14]:
df2.shape

(21, 5)

- 21 municípios goianos possuem taxa de risco calculada pelo INEP